### DISCUSSION TITLE
Radio button that works

### DISCUSSION THREAD

**Discussion by Waldorf-Techie** [kovidgoyal/rapydscript-ng#170] [Ideas]
I am having a problem with Glowscript, which transpiles Python to Javascript through rapydscript.  

Oddly, radio buttons in Glowscript do not function as radio buttons (selecting one does not deselect the others), and it appears this is because the transpiler does not use the built-in Javascript radio button functionality, but instead uses something called _ρσ_interpolate_kwargs_ (???) to generate the button. What _ρσ_interpolate_kwargs_ establishes as a radio button does not operate as one.

I don't know why Rapydscript doesn't just use the Javascript radio button functionality, but there is probably some good reason...

Great program, by the way!
Harlan

Example code produced:
PYTHON CODE:
    radio(bind=R, text='Runradio', name="RDO") 
JAVASCRIPT GENERATED:
    ρσ_interpolate_kwargs.call(this, radio, [ρσ_desugar_kwargs({bind: R, text: "Runradio", name: "RDO"})]);

**Comment by kovidgoyal** [comment]
There is no function named radio(). And interpolate_kwargs is how all
functions with keyword arguments are handled. If you want to create a
radio button in rapydscript you do it exactly like you would in
javascript with docuent.createElement.

**Comment by Waldorf-Techie** [comment]
Thanks.
Do you know why the resulting radio button does not act like a radio
button, i.e. why selecting one does not deselect the others in the names
group?

**Comment by kovidgoyal** [comment]
Read the MDN documentation for radio buttons and how to make a group of
them.

**Comment by Waldorf-Techie** [comment]
I am working in Glowscript.  I cannot determine the transcription of the
radio button, but it doesn’t work properly.

**Comment by Waldorf-Techie** [comment]
Glowscript is a Python implementation that is transpiled to JavaScript.

**Comment by Waldorf-Techie** [comment]
Sorry, to be clear, I am using Glowscript in Vpython mode, as I am teaching
Python at a school.
